src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    public static final double COEFFICIENT = 0.5;
+    private ColorSupplier randomColor = new ColorSupplier();

```

Comment:
Think about which variables should be local in the method and which should be class-level. A ColorSupplier instance should not be a class-level variable in Figure; it should be used where it's needed.